### PR TITLE
Add dummy sources and Gradle profiler

### PR DESCRIPTION
**Pull request by oehme**
This adds some sources to each subproject to make it a little more realistic. It also makes it easier to profile incremental changes to a source file, one of the most important use cases to optimize for.

This currently doesn't build with Buck, it seems that the dexer can't handle the size of the `R` class.

### CODE REVIEW

**Comment by oehme** [comment]
Removed the generated sources and added instructions to the readme.

File: gradle/SourceTemplate/app/build.gradle [hunk 1]
```diff
@@ -0,0 +1,43 @@
+apply plugin: 'com.android.application'
+
+
+buildscript {
+    repositories {
+        jcenter()
+    }
+    dependencies {
+        // other classpath definitions here
+        classpath 'com.neenbedankt.gradle.plugins:android-apt:1.8'
```
Comment (kageiit) [inline gradle/SourceTemplate/app/build.gradle]: should we remove this?

File: gradle/SourceTemplate/app/build.gradle [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    }
+}
+
+android {
+    compileSdkVersion 25
```
Comment (kageiit) [inline gradle/SourceTemplate/app/build.gradle]: these could easily be configured at root project level

File: gradle/SourceTemplate/app/build.gradle (hunk 1 above)
Comment (oehme) [inline gradle/SourceTemplate/app/build.gradle]: Yes, we are using Android's official APT plugin now.

File: gradle/SourceTemplate/app/build.gradle (hunk 2 above)
Comment (oehme) [inline gradle/SourceTemplate/app/build.gradle]: Oh, this is just a harness to make the template build in its own. Only the `src` folder of the template is actually copied to the projects.

File: gradle/SourceTemplate/app/build.gradle (hunk 1 above)
Comment (kageiit) [inline gradle/SourceTemplate/app/build.gradle]: cool. lets remove it :)

File: gradle/SourceTemplate/app/build.gradle (hunk 2 above)
Comment (kageiit) [inline gradle/SourceTemplate/app/build.gradle]: this should be less boiler platey. there is alreadya place available for this

https://github.com/kageiit/android-studio-gradle-test/blob/master/build.gradle#L98

File: gradle/SourceTemplate/app/build.gradle (hunk 1 above)
Comment (oehme) [inline gradle/SourceTemplate/app/build.gradle]: Done

File: gradle/SourceTemplate/app/build.gradle (hunk 2 above)
Comment (oehme) [inline gradle/SourceTemplate/app/build.gradle]: The template project is not part of the build, it is isolated so you can quickly check out whether the sources you write make any sense. Only the `src` dir of the template is actually copied to the main build.

**Pull request merged**